app/models/candidate.rb

```diff
@@ -13,9 +13,13 @@ class Candidate < ApplicationRecord
   has_many :application_forms
 
   def current_application
-    application_form = application_forms.first_or_create
-    # TODO: this is a temporary thing until candidates can choose their course
-    application_form.application_choices.first_or_create(provider_ucas_code: 'ABC')
-    application_form
+    choice = ApplicationChoice.first_or_create do |ac|
```

Comment:
Oops, this fetches the first `ApplicationChoice` (which could belong to a different user, and be completed etc) and then reassigns it further down to belong to the current user's application form.

So:

```ruby
[1] pry(main)> ApplicationForm.all.map(&:destroy!)
[2] pry(main)> ApplicationChoice.all.map(&:destroy!)
[3] pry(main)> ApplicationForm.count
=> 0
[4] pry(main)> ApplicationChoice.count
=> 0
[5] pry(main)> first_user = Candidate.first
[6] pry(main)> second_user = Candidate.second

[7] pry(main)> first_user.current_application
[8] pry(main)> ApplicationForm.count
=> 1 # One more
[9] pry(main)> ApplicationChoice.count
=> 1 # One more
[10] pry(main)> second_user.current_application
[11] pry(main)> ApplicationForm.count
=> 2 # One more
[12] pry(main)> ApplicationChoice.count
=> 1 # Oops!!!
[13] pry(main)> ApplicationChoice.first.application_form.candidate_id == second_user.id
=> true # Oops!!!
```

Reply:
Thanks @tvararu - will get a fix for that raised asap